keras_nlp/models/whisper/whisper_backbone.py

```diff
@@ -0,0 +1,262 @@
+                name=f"transformer_encoder_layer_{i}",
+            )(x)
+
+        x = keras.layers.LayerNormalization(
+            name="encoder_layer_norm",
+            axis=-1,
+            epsilon=1e-5,
+            dtype=tf.float32,
+        )(x)
+        encoder_output = x
+
+        # ====== Decoder ======
+
+        # Embed tokens and positions.
+        x = TokenAndPositionEmbedding(
+            vocabulary_size=vocabulary_size,
+            sequence_length=max_target_sequence_length,
+            embedding_dim=hidden_dim,
+            embeddings_initializer=whisper_kernel_initializer(),
+            name="decoder_token_and_position_embedding",
+        )(decoder_token_id_input)
+
+        # Apply dropout to embeddings.
+        x = keras.layers.Dropout(
+            dropout,
+            name="decoder_embeddings_dropout",
+        )(x)
+
+        # Apply successive transformer decoder blocks.
+        for i in range(num_layers):
+            transformer_decoder_layer = WhisperDecoder(
+                intermediate_dim=intermediate_dim,
+                num_heads=num_heads,
+                dropout=dropout,
+                activation=lambda x: keras.activations.gelu(
+                    x, approximate=False
+                ),
+                layer_norm_epsilon=1e-5,
+                kernel_initializer=whisper_kernel_initializer(),
+                normalize_first=True,
+                name=f"transformer_decoder_layer_{i}",
+                has_cross_attention=True,
+            )
+            x = transformer_decoder_layer(
+                decoder_sequence=x,
+                encoder_sequence=encoder_output,
+                decoder_padding_mask=decoder_padding_mask,
+            )
+
+        x = keras.layers.LayerNormalization(
+            name="decoder_layer_norm",
+            axis=-1,
+            epsilon=1e-5,
+            dtype=tf.float32,
+        )(x)
+        decoder_output = x
+
+        # Instantiate using Functional API Model constructor
+        super().__init__(
+            inputs={
+                "encoder_features": encoder_feature_input,
+                "decoder_token_ids": decoder_token_id_input,
+                "decoder_padding_mask": decoder_padding_mask,
+            },
+            outputs={
+                "encoder_sequence_output": encoder_output,
+                "decoder_sequence_output": decoder_output,
+            },
+            **kwargs,
+        )
+
+        # All references to `self` below this line
+        self.vocabulary_size = vocabulary_size
+        self.num_layers = num_layers
+        self.num_heads = num_heads
+        self.hidden_dim = hidden_dim
+        self.intermediate_dim = intermediate_dim
+        self.num_mels = num_mels
+        self.dropout = dropout
+        self.max_source_sequence_length = max_source_sequence_length
+        self.max_target_sequence_length = max_target_sequence_length
+
+    def get_config(self):
+        return {
```

Comment:
You can chain to super here, and remove `name` and `trainable`. (looks like BART also needs this fix)